packages/block-editor/src/components/block-list/test/block-list-context.native.js

```diff
@@ -0,0 +1,266 @@
+			currentBlockLayouts,
+			NESTED_WITH_INNER_BLOCKS_ID
+		);
+
+		expect( nestedBlock ).toEqual(
+			expect.objectContaining( { clientId: NESTED_WITH_INNER_BLOCKS_ID } )
+		);
+	} );
+
+	it( "finds a deep nested block's layout data", () => {
+		const { findBlockLayoutByClientId } = DEFAULT_BLOCK_LIST_CONTEXT;
+		const currentBlockLayouts = BLOCKS_LAYOUTS_DATA;
+
+		const deepNestedBlock = findBlockLayoutByClientId(
+			currentBlockLayouts,
+			DEEP_NESTED_ID
+		);
+
+		expect( deepNestedBlock ).toEqual(
+			expect.objectContaining( { clientId: DEEP_NESTED_ID } )
+		);
+	} );
+} );
+
+describe( 'deleteBlockLayoutByClientId', () => {
+	it( "deletes a block's layout data at root level", () => {
+		const { findBlockLayoutByClientId } = DEFAULT_BLOCK_LIST_CONTEXT;
+		const defaultBlockLayouts = cloneDeep( BLOCKS_LAYOUTS_DATA );
+		const currentBlockLayouts = deleteBlockLayoutByClientId(
+			defaultBlockLayouts,
+			ROOT_LEVEL_ID
+		);
+
+		const findDeletedBlock = findBlockLayoutByClientId(
+			currentBlockLayouts,
+			ROOT_LEVEL_ID
+		);
+
+		expect( findDeletedBlock ).toEqual(
+			expect.not.objectContaining( { clientId: ROOT_LEVEL_ID } )
+		);
+	} );
+
+	it( "deletes a nested block's layout data with inner blocks", () => {
+		const { findBlockLayoutByClientId } = DEFAULT_BLOCK_LIST_CONTEXT;
+		const defaultBlockLayouts = cloneDeep( BLOCKS_LAYOUTS_DATA );
+		const currentBlockLayouts = deleteBlockLayoutByClientId(
+			defaultBlockLayouts,
+			NESTED_WITH_INNER_BLOCKS_ID
+		);
+
+		const findDeletedBlock = findBlockLayoutByClientId(
+			currentBlockLayouts,
+			NESTED_WITH_INNER_BLOCKS_ID
+		);
+
+		expect( findDeletedBlock ).toEqual(
+			expect.not.objectContaining( {
+				clientId: NESTED_WITH_INNER_BLOCKS_ID,
+			} )
+		);
+	} );
+
+	it( "deletes a deep nested block's layout data", () => {
+		const { findBlockLayoutByClientId } = DEFAULT_BLOCK_LIST_CONTEXT;
+		const defaultBlockLayouts = cloneDeep( BLOCKS_LAYOUTS_DATA );
+		const currentBlockLayouts = deleteBlockLayoutByClientId(
+			defaultBlockLayouts,
+			DEEP_NESTED_ID
+		);
+
+		const findDeletedBlock = findBlockLayoutByClientId(
+			currentBlockLayouts,
+			DEEP_NESTED_ID
+		);
+
+		expect( findDeletedBlock ).toEqual(
+			expect.not.objectContaining( {
+				clientId: DEEP_NESTED_ID,
+			} )
+		);
+	} );
+} );
+
+describe( 'updateBlocksLayouts', () => {
+	it( "adds a new block's layout data at root level im an empty object", () => {
+		const {
+			blocksLayouts,
+			findBlockLayoutByClientId,
+			updateBlocksLayouts,
+		} = DEFAULT_BLOCK_LIST_CONTEXT;
+		const currentBlockLayouts = cloneDeep( blocksLayouts );
+		const BLOCK_CLIENT_ID = PARAGRAPH_BLOCK_LAYOUT_DATA.clientId;
+
+		updateBlocksLayouts( currentBlockLayouts, PARAGRAPH_BLOCK_LAYOUT_DATA );
+
+		const findAddedBlock = findBlockLayoutByClientId(
+			currentBlockLayouts.current,
+			BLOCK_CLIENT_ID
+		);
+
+		expect( findAddedBlock ).toEqual(
+			expect.objectContaining( { clientId: BLOCK_CLIENT_ID } )
+		);
```

Comment:
Maybe we could also check that `rootClientId` is not defined to assure that it was added at root level, wdyt?
```suggestion
		expect( findAddedBlock ).toEqual(
			expect.objectContaining( {
				clientId: BLOCK_CLIENT_ID,
				rootClientId: undefined,
			} )
		);
```

Reply:
Thanks for the suggestion, updated in https://github.com/WordPress/gutenberg/pull/39089/commits/d8f7cf8ada2d91f1233388fbd9f9953791331e37#diff-d3379351048bf4f7ab3045a566f988a3c775892c8c10449fe027c518e08709ebR136